int8[] temp_c
uint16[] vin_mv
